### DISCUSSION TITLE
Accurate Blockplacement not working on fabric 1.19.4 server.

### DISCUSSION THREAD

**Discussion by mkdir-saint52** [maruohon/litematica#788] [Q&A] [answered]
Hi,
I am running a Minecraft fabric 1.19.4 server on Bisect Hosting. I‘ve installed carpet mod, as well as carpet extra. Features like stackable Shulkerboxes work, but when I‘m trying to enable accurateBlockplacement, it does not work. For example when using Litematica, the blocks do nit face the right way. Any ideas? Would appreciate the help :)

**Comment by maruohon** [comment, marked as answer]
If the `accurateBlockPlacement` rule is enabled on the server, then the server side should be fine. If Litematica doesn't automatically detect Carpet mod on the server (which seems to happen sometimes for some people, not sure why), then you may need to change the `easyPlaceProtocolVersion` config from `Auto` to `Version 2`. But remember that you shouldn't leave it at `Version 2` for servers that don't have Carpet Extra and the rule enabled, or for single player (where Litematica would normally use `Version 3` which supports more/all blocks).

Another option you might want to try out is adding the Fabric API mod (I assume yo probably have it already) and the [Litemoretica](https://modrinth.com/mod/litemoretica) mod to both the server and the client. With that mod you get the `Version 3` easy place protocol support, which supports basically all blocks and their properties (that haven't been blacklisted as cheaty).

Later on the v3 support will be directly in Litematica for the client side and in Servux for the server side... but as my progress on the mods has been so slow in the past couple of years, there are now several of these third party mods that add some of these missing things and support for now.

**Reply by mkdir-saint52** [reply]
This worked. Thanks!
